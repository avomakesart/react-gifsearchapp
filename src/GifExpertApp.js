import React, { useState } from 'react'
import AddCategory from './components/AddCategory'
import GifGrid from './components/GifGrid'

export default function GifExpertApp({ defaultCategories = [] }) {
  const [categories, setCategories] = useState(defaultCategories)

  return (
    <>
      <h2>GifExpertApp</h2>
      <AddCategory setCategories={setCategories} />
      <hr className="mt-2" />
      <ol>
        {categories.length > 0 ? (
          categories.map((category) => (
            <GifGrid key={category} category={category} />
          ))
        ) : (
          <h1 className="center mt-10">No hay busquedas</h1>
        )}
      </ol>
    </>
  )
}
