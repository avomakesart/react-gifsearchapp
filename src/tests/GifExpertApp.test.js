import React from 'react'
import '@testing-library/jest-dom'
import GifExpertApp from '../GifExpertApp'
import { shallow } from 'enzyme'

describe('Testing <GifExpertApp />', () => {
  const wrapper = shallow(<GifExpertApp />)

  test('Should show the component correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('should show a categories list', () => {
    const categories = ['Cats', 'Dogs']
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('GifGrid').length).toBe(categories.length)
  })
})
