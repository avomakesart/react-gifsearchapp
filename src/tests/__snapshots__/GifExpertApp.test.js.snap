// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <GifExpertApp /> Should show the component correctly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr
    className="mt-2"
  />
  <ol>
    <h1
      className="center mt-10"
    >
      No hay busquedas
    </h1>
  </ol>
</Fragment>
`;

exports[`Testing <GifExpertApp /> should show a categories list 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr
    className="mt-2"
  />
  <ol>
    <GifGrid
      category="Cats"
      key="Cats"
    />
    <GifGrid
      category="Dogs"
      key="Dogs"
    />
  </ol>
</Fragment>
`;
